os/board/rtl8721csm/hal: enable the loopback mode in UART

deepaksrma opened the pull request.

j11591 committed:
network/wifi : change ndbg->nvdbg for net info log

pillip8282 committed:
security/seclink: add hal test case

add hal test case

sangam-swami committed:
apps/examples/testcase: Fix multiple freeing of buffers

Fix warning for multiple times fd close and freeing of buffers.

Signed-off-by: sangamanatha <sangam.swami@samsung.com>

sangam-swami committed:
apps/example/testcase: Fix warning message for missing % in printf.

Fix warning "printf format string requires 2 parameters but only 1 is given"

Signed-off-by: sangamanatha <sangam.swami@samsung.com>

sunghan-chang committed:
board/rtl8721csm: fix file permission to 644

*.c, *.h and Make* files are readonly files.
Let's set them as 644 file permission.

Signed-off-by: sunghan-chang <sh924.chang@samsung.com>

Commit:
os/driver/lwnl: fix lwnl build error

- fix build error in BLE/WiFi lwnl

Commit:
framework/ble_manager: fix typo

- fix typo

pillip8282 committed:
network/wifi: move wi-fi initialization

move wi-fi initialization to net_initialize

jeongchanKim committed:
include/tinyara/reboot_reason : Rearrange the reboot reason codes

[Reboot Reason Code]
System : 0 ~ 29,
Network : 30 ~ 59,
Common Service : 60 ~ 89,
App : 90 ~ 254

Signed-off-by: jeongchanKim <jc_.kim@samsung.com>

jeongarmy committed:
Add .circleci/config.yml

jeongarmy committed:
CircleCI: Add circleCI for build test

Add config.yml to run circleCI

jeongarmy committed:
travisCI: Remove .travis.yml

Remove .travis.yml because we use circleCI instead of travisCI

pillip8282 committed:
network/lwnl: fix lwnl event

fix lwnl event

pillip8282 committed:
network: fix build error

fix build error

jeongarmy committed:
CircleCI: Separate build jobs for readability

Separate build jobs to show each build result in PR check list

Commit:
os/amebad_ent: fix build error

- fix build error in BLE dev register

pillip8282 committed:
network/wifi: add single scan channel

add single scan channel

AdamZhang0124 committed:
arch/arm/amebad: Fix the 'mixed uart log' issue

This commit fixes the 'mixed uart log' issue which happens when multiple functions use log_uart at the same time.

ziyik committed:
os/board/rtl8721csm: Implement Secure FTL

thapav committed:
apps/examples/testcase/ta_tc/systemio: Fix build breaks when SPI driver TC's are enabled

This patch adds fix for:
1) Missing declarations of SPI driver itc:
	int itc_spi_main(void);
2) Config dependency for unused API's when CONFIG_SPI_EXCHANGE is disabled.
3) SPI bus for UTC to 0.

Signed-off-by: Vidisha Thapa <thapa.v@samsung.com>

thapav committed:
apps/examples/testcase/ta_tc/systemio/itc: Rename API in SPI ITC

Fix typo SPI test case ITc and rename to itc_systemio_spi_write_recv_transfer_p

Signed-off-by: Vidisha Thapa <thapa.v@samsung.com>

an4967 committed:
framework/ble_manager: add check connection function

- add check connection function
- add usage in ble_rmc for example

jeongchanKim committed:
include/tinyara/reboot_reason.h : Adjust reboot reason start code

	/* [Reboot Reason Code]
	 * System : 50 ~ 79(75~79 ; Board Specific),
	 * Network : 80 ~ 109,
	 * Common Service : 110 ~ 139,
	 * App : 140 ~ 254
	 */

Signed-off-by: jeongchanKim <jc_.kim@samsung.com>

ziyik committed:
os/arch/arm/src/amebad: Add all RTK reboot reason

jeongchanKim committed:
os/include/tinyara/reboot_reason.h : Change board specific reboot reason from 250.

For board specific reboot reason, change to start them from 250.

Signed-off-by: jeongchanKim <jc_.kim@samsung.com>

Commit:
framework/ble_manager: separate ble_manager init process

- The initialization is separated into 2 parts, server & client

jeongchanKim committed:
build/configs : Change lpwork priority as 50

Change lpwork priority as 50, but except for artik05x because artik05x is designed to use lpwork prio as 176.

Signed-off-by: jeongchanKim <jc_.kim@samsung.com>

Commit:
framework/ble_manager: apply lwnl in BLE callback process

- change BLE callback process from direct call to lwnl handling.
- re-order mac address of scanned data.

Commit:
drivers/lwnl: apply lwnl bind feature.

- apply lwnl BLE & WiFi addr binding feature.

pillip8282 committed:
network/netmgr: add event handler task

add event handler task not to driver create new thread if wi-fi event occurs.
	/* Allocate a detachable structure to support pthread_join logic */

	pjoin = (FAR struct join_s *)kmm_zalloc(sizeof(struct join_s));
	if (!pjoin) {
		sdbg("ERROR: Failed to allocate join\n");
		errcode = ENOMEM;
		goto errout_with_tcb;
	}

0  kmm_zalloc (size=32) at kmm_heap/kmm_zalloc.c:120
1  0x041798a4 in pthread_create (thread=0x2024540 <g_dhcpd_tid>, attr=0x2063c38,
   start_routine=0x41961c4 <_dhcpd_join_handler>, arg=0x205fd30) at pthread/pthread_create.c:315
2  0x0419659c in dhcp_server_start (intf=0x41aa560 "wlan0",
   dhcp_join_cb=0x4174efc <_wifi_dhcps_event>) at dhcpd_lwip.c:343
3  0x04175440 in wm_dhcps_start (cb=0x4174efc <_wifi_dhcps_event>)
   at src/wifi_manager/wifi_manager_dhcps.c:156
4  0x0417418c in _wifimgr_run_softap (config=0x206795c)
   at src/wifi_manager/wifi_manager_state.c:281
5  0x041744b4 in _handler_on_disconnected_state (msg=0x2067930)
   at src/wifi_manager/wifi_manager_state.c:375
6  0x04174ee4 in wifimgr_handle_request (msg=0x2067930)
   at src/wifi_manager/wifi_manager_state.c:627
7  0x04173964 in wifimgr_message_out (msg=0x2063d8c, queue=0x2027950 <g_wifi_message_queue>)
   at src/wifi_manager/wifi_manager_message.c:120
8  0x041729e0 in _process_msg (argc=1, argv=0x2063dbc)
    at src/wifi_manager/wifi_manager_msghandler.c:52

sungkyu0 committed:
file : Add format functionality for both of internal & external

Add format feature and tash command
Format works only with blockdriver which is registered for smartfs

an4967 committed:
framework/ble_manager: refactor functions of 'bonded device'

- integrate 'bonded device' control functions.
- add examples of 'bonded device' control functions.

ziyik committed:
os/board/rtl8721csm: Enable RTK debug message

Commit:
net/blemgr/interface: add 'ble_addr' struct

- add 'ble_addr' struct
- apply 'ble_addr' to ble_drv / ble stack

Commit:
framework/ble_manager: add 'set_adv_type' api

- add 'set_adv_type' api
- apply 'ble_addr' struct

pillip8282 committed:
driver/wireless/virtual: fix memory leak

fix memory leak in virtual mock driver.

pillip8282 committed:
framework/wifi_manager: add checking channel range

add checking channel range

jeongchanKim committed:
arch/arm/src/armv8-m/up_svcall.c : Add missing TZ_LoadContext_S when restore context

Before restoring the tcb context, we should load the secure context of next running tcb.

jeongarmy committed:
apps/examples/timer: Close fd before pthread exit to release used resource

It should close fd before pthread_exit to release used resource.
If timer example runs continuously, an error occurs in 4 times.

TASH>>timer -n 10

Open /dev/timer0
Attach timer handler
Set timer interval to 1000000, repeat 10
Start the timer
FRT Open /dev/timer1
ERROR: Failed to open Free Run Timer: 24
Finished

pillip8282 committed:
apps/wm_test: add scan and join performance tool

add scan and join performance tool

jeongchanKim committed:
os/include/tinyara/lwnl/lwnl.h : Add header for bool and uint type

Add stdbool.h and stdint.h for build error

Signed-off-by: jeongchanKim <jc_.kim@samsung.com>

jeongchanKim committed:
external/iotivity/caipserver.c : Add lwnl.h include

For using LWNL_DEV_WIFI, lwnl.h is required.

Signed-off-by: jeongchanKim <jc_.kim@samsung.com>

kishore-sn committed:
testcase/le_tc/network: fix build errors on tc_net_pbuf

Fix following build errors:
CC:  le_tc/network/tc_net_pbuf.c
le_tc/network/tc_net_pbuf.c: In function 'tc_net_pbuf_layer_ip_p':
le_tc/network/tc_net_pbuf.c:76:36: warning: passing argument 3 of 'ioctl' makes integer from pointer without a cast [-Wint-conversion]
   76 |  ret = ioctl(fd, TESTIOC_NET_PBUF, &args);
      |                                    ^~~~~
      |                                    |
      |                                    struct pbuf_test_args *
In file included from le_tc/network/tc_net_pbuf.c:32:
/home/kishore/tinyara/TizenRT_srib_gerrit/os/include/sys/ioctl.h:140:42: note: expected 'long unsigned int' but argument is of type 'struct pbuf_test_args *'
  140 | int ioctl(int fd, int req, unsigned long arg);
      |                            ~~~~~~~~~~~~~~^~~
le_tc/network/tc_net_pbuf.c: In function 'tc_net_pbuf_layer_link_p':
le_tc/network/tc_net_pbuf.c:98:36: warning: passing argument 3 of 'ioctl' makes integer from pointer without a cast [-Wint-conversion]
   98 |  ret = ioctl(fd, TESTIOC_NET_PBUF, &args);
      |                                    ^~~~~
      |                                    |
      |                                    struct pbuf_test_args *
In file included from le_tc/network/tc_net_pbuf.c:32:
/home/kishore/tinyara/TizenRT_srib_gerrit/os/include/sys/ioctl.h:140:42: note: expected 'long unsigned int' but argument is of type 'struct pbuf_test_args *'
  140 | int ioctl(int fd, int req, unsigned long arg);
      |                            ~~~~~~~~~~~~~~^~~
le_tc/network/tc_net_pbuf.c: In function 'tc_net_pbuf_layer_raw_tx_p':
le_tc/network/tc_net_pbuf.c:120:36: warning: passing argument 3 of 'ioctl' makes integer from pointer without a cast [-Wint-conversion]
  120 |  ret = ioctl(fd, TESTIOC_NET_PBUF, &args);
      |                                    ^~~~~
      |                                    |
      |                                    struct pbuf_test_args *
In file included from le_tc/network/tc_net_pbuf.c:32:
/home/kishore/tinyara/TizenRT_srib_gerrit/os/include/sys/ioctl.h:140:42: note: expected 'long unsigned int' but argument is of type 'struct pbuf_test_args *'
  140 | int ioctl(int fd, int req, unsigned long arg);
      |                            ~~~~~~~~~~~~~~^~~
le_tc/network/tc_net_pbuf.c: In function 'tc_net_pbuf_layer_raw_p':
le_tc/network/tc_net_pbuf.c:142:36: warning: passing argument 3 of 'ioctl' makes integer from pointer without a cast [-Wint-conversion]
  142 |  ret = ioctl(fd, TESTIOC_NET_PBUF, &args);
      |                                    ^~~~~
      |                                    |
      |                                    struct pbuf_test_args *
In file included from le_tc/network/tc_net_pbuf.c:32:
/home/kishore/tinyara/TizenRT_srib_gerrit/os/include/sys/ioctl.h:140:42: note: expected 'long unsigned int' but argument is of type 'struct pbuf_test_args *'
  140 | int ioctl(int fd, int req, unsigned long arg);
      |                            ~~~~~~~~~~~~~~^~~
le_tc/network/tc_net_pbuf.c: In function 'tc_net_pbuf_layer_n':
le_tc/network/tc_net_pbuf.c:164:36: warning: passing argument 3 of 'ioctl' makes integer from pointer without a cast [-Wint-conversion]
  164 |  ret = ioctl(fd, TESTIOC_NET_PBUF, &args);
      |                                    ^~~~~
      |                                    |
      |                                    struct pbuf_test_args *
In file included from le_tc/network/tc_net_pbuf.c:32:
/home/kishore/tinyara/TizenRT_srib_gerrit/os/include/sys/ioctl.h:140:42: note: expected 'long unsigned int' but argument is of type 'struct pbuf_test_args *'
  140 | int ioctl(int fd, int req, unsigned long arg);
      |                            ~~~~~~~~~~~~~~^~~
le_tc/network/tc_net_pbuf.c: In function 'tc_net_pbuf_type_n':
le_tc/network/tc_net_pbuf.c:186:36: warning: passing argument 3 of 'ioctl' makes integer from pointer without a cast [-Wint-conversion]
  186 |  ret = ioctl(fd, TESTIOC_NET_PBUF, &args);
      |                                    ^~~~~
      |                                    |
      |                                    struct pbuf_test_args *
In file included from le_tc/network/tc_net_pbuf.c:32:
/home/kishore/tinyara/TizenRT_srib_gerrit/os/include/sys/ioctl.h:140:42: note: expected 'long unsigned int' but argument is of type 'struct pbuf_test_args *'
  140 | int ioctl(int fd, int req, unsigned long arg);
      |                            ~~~~~~~~~~~~~~^~~

Signed-off-by: Kishore S N <kishore.sn@samsung.com>

sangam-swami commented on apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:
```diff
@@ -360,8 +360,8 @@ static void itc_systemio_iotbus_uart_write_read_flush_p(void)
-	char sz_input_text1[BUF_LEN] = "UART READ/WRITE ITC TESTING!";
-	char sz_input_text2[BUF_LEN] = "UART READ/WRITE ITC TESTING1!";
+	char sz_input_text1[BUF_LEN] = "UART READ/WRITE";
+	char sz_input_text2[BUF_LEN] = "UART READ/WRITE1";
```
make it uniform WRITE2

deepaksrma replied:
corrected

sangam-swami commented on apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:
```diff
@@ -636,7 +638,7 @@ static void itc_systemio_iotbus_uart_set_mode_flush_write_read_p(void)
 	int i_stop_bits = 1;
 	int ret = IOTBUS_ERROR_NONE;
-	char sz_input_text[BUF_LEN] = "UART READ/WRITE ITC TESTING!";
+	char sz_input_text[BUF_LEN] = "UART READ/WRITE";
```
UART READ/WRITE TESTING

deepaksrma replied:
corrected

sangam-swami commented on lib/libc/termios/lib_loopback_en.c:
```diff
@@ -0,0 +1,72 @@
+ *   calls the ioctl to enable/ disable the loopback mode of UART
+ *
+ * Input Parameters:
+ *   fd  - The 'fd' argument is an open file descriptor associated with a terminal.
```
struct termios also you can add in input list

deepaksrma replied:
corrected

sangam-swami commented on lib/libc/termios/lib_loopback_en.c:
```diff
@@ -0,0 +1,72 @@
+
+#include <termios.h>
+#include <errno.h>
+
```
all these headers needed?

deepaksrma replied:
corrected

Commit:
framework/src/iotbus: changes in iotbus to add loopback mode in UART.

- add the changes in iotbus to support loopback mode in UART.

Signed-off-by: Deepak Sharma <deepak.sh@samsung.com>

Commit:
os/board/rtl8721csm/hal: add the loopback mode for UART

- add the loopback mode to test UART TCs.

Signed-off-by: Deepak Sharma <deepak.sh@samsung.com>

Commit:
apps/examples/testcase/systemio/itc: refactor the itc UART TCs

- Reduce the input string length for write and read operations because tx fifo and rx fifo size is 16 bytes,
  so if string length is more than 16 bytes, in loopback mode, extra bytes cause the mismatch error between input and output string.

Signed-off-by: Deepak Sharma <deepak.sh@samsung.com>

sangam-swami committed:
os/kernel/debug: Remove wrong description of sysdbg_read function

Remove wrong description of sysdbg_read function

Signed-off-by: sangamanatha <sangam.swami@samsung.com>
